url2a908001cd_malictus_klang_tgz-pJ8-malictus_klang_test_PrimitiveTestJ8/src/malictus/klang/chunk/PlainTextChunk.java

```diff
@@ -0,0 +1,115 @@
+		super(parentFile, startpos, endpos, datastartpos, dataendpos, chunkType, chunkName, parentChunk, raf);
+		this.textEncoding = textEncoding;
+		raf.seek(datastartpos);
+		/*
+		 * LOAD PRIMITIVE OBJECTS (UNINITIALIZED)
+		 */
+		/*
+		 * READ FROM FILE TO FILL PRIMITIVE DATA
+		 */
+		PlainString val  = null;
+		try {
+			long bytepos = raf.getFilePointer();
+			val = new PlainString(raf, dataendpos, textEncoding, true);
+			PrimitiveData valData = new PrimitiveData(val, KlangConstants.PRIMITIVE_DATA_PLAINSTRING_CHUNK_VALUE, bytepos, raf.getFilePointer());
+			/*
+			 * ADD PRIMITIVES TO VECTOR
+			 */
+			this.primitives.add(valData);
+		} catch (BadValueException er) {
+			er.printStackTrace();
+			try {
+				val = new PlainString("", textEncoding, true);
+			} catch (BadValueException e) {
+				throw new IOException();
+			}
+		}	
+	}
+	
+	public void setTextEncoding(String newEncoding) {
+		this.textEncoding = newEncoding;
+	}
+	
+	public String getTextEncoding() {
+		return textEncoding;
+	}
+	
+	/**
+	 * Overwrite of this method in Chunk class to write string.
+	 * @param raff a pointer to the file (position is not important)
+	 * @throws IOException if any read/write error occurs
+	 */
+	public void reparseChunkPrimitives(RandomAccessFile raff) throws IOException {
+		if ((this.primitives == null) || (this.primitives.size() == 0)) {
+			//nothing to do!
+			return;
+		}
+		Primitive p = primitives.get(0).getPrimitive();		
+		long oldsize = this.getDataEndPosition() - this.getDataStartPosition();
+		long pos = 0;
+		try {
+			long newsize = p.writeValueToString().getBytes(textEncoding).length;
+			boolean needsPad = false;
+			if (this.getParentChunk() != null) {
+				if (this.getParentChunk().usesPadByte()) {
+					newsize = KlangUtil.adjustForPadByte(newsize);
+					needsPad = true;
+				}
+			}
+			long totnewsize = newsize + (this.getEndPosition() - this.getStartPosition()) - (this.getDataEndPosition() - this.getDataStartPosition());
+			if (newsize < oldsize) {
+				this.chunkIsAboutToChangeSize(totnewsize, raff);
+				KlangUtil.deleteFromFile(this.getParentFile().getFile(), this.getDataStartPosition(), this.getDataStartPosition() + (oldsize - newsize));	
+			} else if (newsize > oldsize) {
+				this.chunkIsAboutToChangeSize(totnewsize, raff);
+				int amt = (int)(newsize - oldsize);
+				byte[] newbyte = new byte[amt];
+				KlangUtil.insertIntoFile(newbyte, this.getParentFile().getFile(), this.getDataStartPosition());
+			}
+			//write null to pad byte
+			if (needsPad) {
+				raff.seek(this.getDataStartPosition() + newsize - 1);
+				raff.write(0);
+			}
+			raff.seek(this.getDataStartPosition());
+			pos = raff.getFilePointer();
+			//write value back to the file
+			p.writeValueToFile(raff);
+		} catch (BadValueException err) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)
Similar issue at line numbers 105.